src/overlays/actors/ovl_En_Am/z_en_am.c

```diff
@@ -93,63 +112,881 @@ static ColliderQuadInit D_809AFFD8 = {
+                EnAm_SetupRotateToInit(this);
+            }
+        }
+    }
+
+    // turn away from a wall if touching one
+    if ((this->dyna.actor.speedXZ != 0.0f) && (this->dyna.actor.bgCheckFlags & 8)) {
+        this->dyna.actor.world.rot.y = this->dyna.actor.wallYaw;
+        Actor_MoveForward(&this->dyna.actor);
+    }
+
+    SkelAnime_Update(&this->skelAnime);
+
+    this->dyna.actor.shape.rot.y = this->dyna.actor.world.rot.y;
+}
+
+void EnAm_RecoilFromDamage(EnAm* this, GlobalContext* globalCtx) {
+    if (this->dyna.actor.speedXZ < 0.0f) {
+        this->dyna.actor.speedXZ += 0.5f;
+    }
+
+    if ((this->dyna.actor.velocity.y <= 0.0f) && !EnAm_CanMove(this, globalCtx, -8.0f, this->dyna.actor.world.rot.y)) {
+        this->dyna.actor.speedXZ = 0.0f;
+    }
+
+    if (SkelAnime_Update(&this->skelAnime)) {
+        EnAm_SetupLunge(this);
+        this->deathTimer = 64;
+        this->panicSpinRot = 0;
+    }
+}
+
+/**
+ * After doing 3 lunges, wait for 2 seconds before attacking again.
+ * Turn toward the player before lunging again.
+ */
+void EnAm_Cooldown(EnAm* this, GlobalContext* globalCtx) {
+    s16 yawDiff = this->dyna.actor.yawTowardsPlayer - this->dyna.actor.world.rot.y;
+
+    if (yawDiff < 0) {
+        yawDiff = -yawDiff;
+    }
+
+    if (this->cooldownTimer != 0) {
+        this->cooldownTimer--;
+    } else {
+        if (this->skelAnime.curFrame == 8.0f) {
+            Math_SmoothStepToS(&this->dyna.actor.world.rot.y, this->dyna.actor.yawTowardsPlayer, 1, 0x1F40, 0);
+            this->dyna.actor.velocity.y = 12.0f;
+        } else if (this->skelAnime.curFrame > 11.0f) {
+            if (!(this->dyna.actor.bgCheckFlags & 1)) {
+                this->skelAnime.curFrame = 11;
+            } else {
+                if (yawDiff < 3500) {
+                    this->unk_258 = 0;
+                }
+                this->dyna.actor.velocity.y = 0.0f;
+                this->dyna.actor.world.pos.y = this->dyna.actor.floorHeight;
+                EnAm_SpawnEffects(this, globalCtx);
+            }
+        }
+
+        SkelAnime_Update(&this->skelAnime);
+
+        if (this->unk_258 == 0) {
+            EnAm_SetupLunge(this);
+            Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EN_AMOS_VOICE);
+        }
+
+        this->dyna.actor.shape.rot.y = this->dyna.actor.world.rot.y;
+    }
+}
+
+/**
+ * Lunge toward the player in an attempt to deal damage. Hop 3 times.
+ * Used for both normal attacks and the death sequence.
+ */
+void EnAm_Lunge(EnAm* this, GlobalContext* globalCtx) {
+    if (this->deathTimer < 52) {
+        if (this->skelAnime.curFrame == 8.0f) {
+            this->dyna.actor.velocity.y = 12.0f;
+
+            if (EnAm_CanMove(this, globalCtx, 80.0f, this->dyna.actor.world.rot.y)) {
+                this->dyna.actor.speedXZ = 6.0f;
+            } else {
+                this->dyna.actor.speedXZ = 0.0f;
+            }
```

Comment:
Does this work as a ternary? Does it look better as one?

Reply:
imo no it doesnt look better, line is very long. doesnt quite hit the line limit but still doesnt look as good